src/server/journal/journal_test.cc

```diff
@@ -125,5 +126,74 @@ TEST(Journal, WriteRead) {
   }
 }
 
+TEST(Journal, PendingBuf) {
+  PendingBuf pbuf;
+
+  ASSERT_TRUE(pbuf.empty());
+  ASSERT_EQ(pbuf.size(), 0);
+
+  pbuf.push("one");
+  pbuf.push(" small");
+  pbuf.push(" test");
+
+  ASSERT_FALSE(pbuf.empty());
+  ASSERT_EQ(pbuf.size(), 14);
+
+  {
+    auto& sending_buf = pbuf.PrepareSendingBuf();
+    ASSERT_EQ(sending_buf.buf.size(), 3);
+    ASSERT_EQ(sending_buf.mem_size, 14);
+
+    ASSERT_EQ(sending_buf.buf[0], "one");
+    ASSERT_EQ(sending_buf.buf[1], " small");
+    ASSERT_EQ(sending_buf.buf[2], " test");
+  }
+
+  const size_t string_num = 2000;
```

Comment:
Please add
`EXPECT_LT(string_num, PendingBuf::Buf::max_buf_size);`
Then below use `string_num` or `PendingBuf::Buf::max_buf_size` directly, instead of for example `std::min()`

Reply:
UIO_MAXIOV can be more than 1024 so I would prefer to leave it as is, and we will test different scenarios on the different platforms

Reply:
I've read your next comment and I think I know how to rewrite it